### PR TITLE
EPMRPP-38165 Log page. New attachments carousel. bugfix/EPMRPP-38344

### PR DESCRIPTION
**Pull request by AmsterGet**
Task - https://jiraeu.epam.com/browse/EPMRPP-38165
Carousel package link - https://www.npmjs.com/package/pure-react-carousel
Bug - https://jiraeu.epam.com/browse/EPMRPP-38344

### CODE REVIEW

File: app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx [hunk 1]
```diff
@@ -22,56 +26,134 @@ const cx = classNames.bind(styles);
...
+    const { activeItemId, isMobileView, pagination } = props;
+
+    const visibleThumbs = Attachments.getVisibleThumbs(isMobileView);
+    const currentThumb = activeItemId
+      ? Math.floor(activeItemId / visibleThumbs) * visibleThumbs
+      : 0;
+
+    this.state = {
+      mainAreaVisible: activeItemId !== null,
+      activeItemId,
```
Comment (margalencia) [inline app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx]: It looks a little bit strange, you receive `activeItemId` as a prop, and then change it inside this component (update its value in the state) and not update `activeItemId` in the outer component. Is this behavior correct?

Comment (AmsterGet) [reply app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx]: I update it in the outer component also. F.e. looks on [this](https://github.com/reportportal/service-ui/pull/1300/files/467eaa13d0692d965e2341b47021bbc8f44ecfa1#diff-48d4e51950475e052cf1f5f5eef6f703R103) line

Comment (margalencia) [reply app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx]: So, in this case, my do you need to save it in the state? Maybe you can just use it directly from props?

Comment (AmsterGet) [reply app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx]: It was a problem with the active tab content rendering process in `infoTabs`.
Now fixed and `activeItemId` uses directly from props.

File: app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx [hunk 2]
```diff
@@ -28,56 +32,132 @@ const cx = classNames.bind(styles);
...
+        currentThumb: thumbConfig ? thumbConfig.currentThumb : this.state.currentThumb,
+      },
+      () => this.loadNewItems(prevActiveItemId, thumbConfig),
+    );
+  };
+
+  loadNewItems = (prevActiveItemId, thumbConfig) => {
+    if (thumbConfig && this.props.activeItemId > prevActiveItemId) {
+      const currentPage = Math.floor(thumbConfig.currentThumb / this.state.size) + 1;
+      currentPage + 1 >= this.state.page && this.handleNewAttachmentsLoad();
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx]: Fixed

File: app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx [hunk 3]
```diff
@@ -28,56 +32,132 @@ const cx = classNames.bind(styles);
...
     }
-    if (!this.state.mainAreaVisible) {
-      this.setState({ mainAreaVisible: true });
+  };
+
+  handleNewAttachmentsLoad = () => {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/attachments/attachments.jsx]: Fixed

**Pull request merged**